drivers/video/mipidsi/mipi_dsi_device.c

```diff
@@ -0,0 +1,1023 @@
+ *   len - command payload length
+ *
+ * Returned Value:
+ *   The number of bytes successfully transmitted or a negative error
+ *   code on failure.
+ *
+ ****************************************************************************/
+
+ssize_t mipi_dsi_dcs_write(FAR struct mipi_dsi_device *device,
+                           uint8_t cmd,
+                           FAR const void *data,
+                           size_t len)
+{
+  ssize_t ret;
+  uint8_t stack_tx[8];
+  FAR uint8_t *tx = stack_tx;
+
+  if (len > sizeof(stack_tx) - 1)
+    {
+      tx = kmm_malloc(len + 1);
+      if (!tx)
+        {
+          return -ENOMEM;
+        }
+    }
+
+  /* concatenate the DCS command byte and the payload */
+
+  tx[0] = cmd;
+  if (data)
+    {
+      memcpy(&tx[1], data, len);
+    }
+
+  ret = mipi_dsi_dcs_write_buffer(device, tx, len + 1);
+
+  if (tx != stack_tx)
+    {
+      kmm_free(tx);
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: mipi_dsi_dcs_read
+ *
+ * Description:
+ *   Send DCS read request command.
+ *
+ * Input Parameters:
+ *   device - DSI peripheral device
+ *   cmd - DCS command
+ *   data - buffer in which to receive data
+ *   len - size of receive buffer
+ *
+ * Returned Value:
+ *   The number of bytes read or a negative error code on failure.
+ *
+ ****************************************************************************/
+
+ssize_t mipi_dsi_dcs_read(FAR struct mipi_dsi_device *device,
+                          uint8_t cmd,
+                          FAR void *data,
+                          size_t len)
+{
+  struct mipi_dsi_msg msg =
+  {
+    .channel = device->channel,
+    .type = MIPI_DSI_DCS_READ_0_PARAM,
+    .tx_buf = &cmd,
+    .tx_len = 1,
+    .rx_buf = data,
+    .rx_len = len
+  };
+
+  return mipi_dsi_transfer(device, &msg);
+}
+
+/****************************************************************************
+ * Name: mipi_dsi_dcs_nop
+ *
+ * Description:
+ *   Send DCS nop packet
+ *
+ * Input Parameters:
+ *   device - DSI peripheral device
+ *
+ * Returned Value:
+ *   OK on success or a negative error code on failure.
+ *
+ ****************************************************************************/
+
+int mipi_dsi_dcs_nop(FAR struct mipi_dsi_device *device)
+{
+  int ret;
+
+  ret = mipi_dsi_dcs_write(device, MIPI_DCS_NOP, NULL, 0);
+  return ret < 0 ? ret : OK;
+}
+
+/****************************************************************************
+ * Name: mipi_dsi_dcs_soft_reset
+ *
+ * Description:
+ *   Send a software reset of the display module
+ *
+ * Input Parameters:
+ *   device - DSI peripheral device
+ *
+ * Returned Value:
+ *   OK on success or a negative error code on failure.
+ *
+ ****************************************************************************/
+
+int mipi_dsi_dcs_soft_reset(FAR struct mipi_dsi_device *device)
+{
+  int ret;
+
+  ret = mipi_dsi_dcs_write(device, MIPI_DCS_SOFT_RESET, NULL, 0);
+
```

Comment:
some similar places have new line while other do not. Maybe some common style can be used across this file?